pkg/pgmodel/ingestor/duplicates.go

```diff
@@ -0,0 +1,42 @@
+package ingestor
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/pgmodel/metrics"
+)
+
+const reportDuplicatesInterval = time.Minute
+
+var (
+	launchReporterOnce    sync.Once
+	duplicateMetricsTotal atomic.Value
+)
+
+func init() {
+	duplicateMetricsTotal.Store(uint64(0))
+}
+
+func registerDuplicates(duplicateSamples int64) {
+	metrics.DuplicateSamples.Add(float64(duplicateSamples))
+	metrics.DuplicateWrites.Inc()
+}
+
+func reportDuplicates(duplicateMetrics uint64) {
+	duplicateMetricsTotal.Store(duplicateMetricsTotal.Load().(uint64) + duplicateMetrics)
```

Comment:
this is the wrong way to use atomics. This isn't concurrency safe (if two goroutines call this at the same time, the last one wins, ignoring the first addition),  Instead you should be using
```suggestion
	atomic.AddUint64(&duplicateMetricsTotal, duplicateMetrics)
```

Reply:
Oh yes, sorry I took this as the uber's atomic package usage way.